### PR TITLE
fix: vue2 global directives in component testing

### PR DESCRIPTION
**Pull request by crazyk2**
### User facing changelog
`mount()` now accept a extensions.directives option, which will be registered globally for Vue2 component testing

### Additional details
To register custom vue directive you should pass option like
```js
    mount(HelloWorld, {
      directives: { customDirective: { /* stub */ }}
    })
```
and this works but only for first level mounted component.

If you test component with nested components which have custom directive, you see error in console
`[Vue warn]: Failed to resolve directive: customDirective`

### How has the user experience changed?
Now you can pass 
```js
  mount(HelloWorld, {
    extensions: {
      directives: { customDirective: { /* stub */ }}
    },
  })
```
And it will be registered globally and works everywhere.

See example /system-tests/projects/vueclivue2-configured/src/components/HelloWorld.cy.js

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by lmiller1990 — APPROVED** [review]
Looks good... I guess this is a bug in Test Utils, I wonder if we should patch it there, too. Happy to do it here, though - thanks for the PR and test.

File: system-tests/projects/vueclivue2-configured/src/components/HelloWorld.cy.js [hunk 1]
```diff
@@ -13,4 +15,30 @@ describe('<Logo />', () => {
 
     cy.contains('h1', slotContent)
   })
+
+  it('Vue2 custom directive should work ', () => {
```
Comment (lmiller1990) [inline system-tests/projects/vueclivue2-configured/src/components/HelloWorld.cy.js:28]: Nice job figuring out where to add a test 👍

**Comment by lmiller1990** [comment]
Hmmm looks like some CI failures. I will take a look.

**Comment by lmiller1990** [comment]
I pushed a commit to fix the tests.

**Review by rockindahizzy — APPROVED** [review]

**Pull request merged**